src/chain_parsers/visualsign-ethereum/src/contracts/core/erc721.rs

```diff
@@ -0,0 +1,72 @@
+//! ERC-721 NFT Standard Visualizer
+//!
+//! Provides visualization for common ERC-721 functions.
+//!
+//! Reference: <https://eips.ethereum.org/EIPS/eip-721>
+
+#![allow(unused_imports)]
+
+use alloy_sol_types::{SolCall, sol};
+use visualsign::{SignablePayloadField, SignablePayloadFieldCommon, SignablePayloadFieldTextV2};
+
+// ERC-721 interface
+sol! {
+    interface IERC721 {
+        function balanceOf(address owner) external view returns (uint256 balance);
+        function ownerOf(uint256 tokenId) external view returns (address owner);
+        function safeTransferFrom(address from, address to, uint256 tokenId) external;
+        function safeTransferFrom(address from, address to, uint256 tokenId, bytes calldata data) external;
+        function transferFrom(address from, address to, uint256 tokenId) external;
+        function approve(address to, uint256 tokenId) external;
+        function setApprovalForAll(address operator, bool approved) external;
+        function getApproved(uint256 tokenId) external view returns (address operator);
+        function isApprovedForAll(address owner, address operator) external view returns (bool);
+    }
+}
+
+/// Visualizer for ERC-721 NFT contract calls
+pub struct ERC721Visualizer;
+
+impl ERC721Visualizer {
+    /// Attempts to decode and visualize ERC-721 function calls
+    ///
+    /// # Arguments
+    /// * `input` - The calldata bytes
+    ///
+    /// # Returns
+    /// * `Some(field)` if a recognized ERC-721 function is found
+    /// * `None` if the input doesn't match any ERC-721 function
+    pub fn visualize_tx_commands(&self, input: &[u8]) -> Option<SignablePayloadField> {
+        if input.len() < 4 {
+            return None;
+        }
+
```

Comment:
What's the thinking here? We return none anyway (below). Does this input.len() < 4 map to a well-known error case?

Reply:
In the test it seems like it is ("too short"). I worry that this interface (`Option`) is confusing, because it'll be hard to distinguish "not supported" vs "error". I'd recomment switching to `Result<SignablePayloadField, ErrorType>`. One of the error variant can then be "Unsupported", and the other can be "InputTooShort".